### ISSUE TITLE
YouTube Music Doesn't Automatically Skip Disliked Songs

### ISSUE THREAD

**Issue by andrewjmetzger** [MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-#3878]
**Checklist:**

* [x] I have checked that there are no issues with similar or the same content

  **YOU SHOULD CHECK CLOSED ISSUES AS WELL**
* [x] I have checked the FAQ (https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/wiki/FAQ) and the answer I am looking for is not there
* [x] I have double checked and can reproduce the issue

**OS:** Pop!_OS 20.04 LTS

**GPMDP Version:** 4.7.1 (deb)

**Issue Descriptions:** In YouTube Music mode, disliked songs are never skipped during playback. 

**Steps to Reproduce:**

1. Switch to YouTube Music mode using the button in the upper-right, or the tray icon.
1. Using the tray icon, open **Desktop settings**.
1. Select the **Playback.** tab, and ensure **Automatically skip disliked songs** is checked.
1. Queue any song, followed by a disliked song, and then another song.
1. Result: All three songs are played.

Expected behavior: Songs that are liked or have no rating play. Songs that are disliked do not play.

What actually happens: All songs play, even if disliked or rated with a thumbs down.

**Comment by CrankySupertoon** [comment]
Simple answer. This is google play music desktop player not youtube music desktop player. Try this

https://ytmdesktop.app/

**Comment by andrewjmetzger** [comment]
Thanks for the link. However, [GPMDP also supports YTM](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/issues/3841).

Until a maintainer closes YTM issues as "wontfix" or decides to drop YTM support altogether, issues like this are a valid concern for GPMDP, despite the name of the project.

See also: Issue #3841.

**Comment by andrewjmetzger** [comment]
@CrankySupertoon YTM Desktop, which you linked above, does not currently support the desired feature. GPMDP doe support this with GPM, but not for YTM. When you suggest an alternative, please make sure it is a comparable replacement.

**Comment by returnvisit** [comment]
me too. In YouTube Music mode, disliked songs are never skipped during playback

**Comment by Block81838** [comment]
> Simple answer. This is google play music desktop player not youtube music desktop player. Try this
> 
> https://ytmdesktop.app/

This is freakin lag, a real resource killer

**Comment by andrewjmetzger** [comment]
@Block81838 Agreed, although I don't see any feature-comparable alternatives, since the official site on desktop can *also* be a resource killer. With development of GPMDP all but ceased, it's not like we have many options. 

As time goes on, I'm sure YTM Desktop will continue to improve. For now, I'd suggest pulling [the development branch](https://github.com/ytmdesktop/ytmdesktop/tree/development), where all the latest improvements are happening.

**Comment by Block81838** [comment]
@andrewjmetzger In fact, I prefer to use gpmdp, it is smooth and full-featured. And gpmdp is based on gtk, ytm desktop is based on the electron, on the word, electron is a resource killer.
gpmdp is enough for me (gtk is always the best choice)

**Comment by Block81838** [comment]
@andrewjmetzger does gpmdp no longer develop?

**Comment by andrewjmetzger** [comment]
Agreed, Electron is a resource killer. If I knew how, I would update GPMDP to be YTM Desktop 2.0 (or similar) and remove the GPM branding to breathe some life back into this project so that we could avoid electron altogether. Alas, I have no clue how to do any of that.

> does gpmdp no longer develop?

Since Google Play Music is shutting down [at the end of October 2020](https://arstechnica.com/gadgets/2020/10/youtube-music-opens-up-google-cast-support-for-everyone/), this project needs a major update to shift its support to YouTube Music. Unfortunately, GPMDP's [commit history](https://github.com/MarshallOfSound/Google-Play-Music-Desktop-Player-UNOFFICIAL-/commits/master) shows that the last commit on 2020-08-10 was a translation patch--and nothing new has been added been added since July. 

Several issues have been raised to discuss the future of this project, but with no clear word from the project maintainers as to the path forward for GPMDP, it seems that YTM Desktop is the least worst replacement for now.

**Comment by christofdamian** [comment]
Having just had to fix an issue in ytmdesktop I would be happy to see GPMDP remove Google Music and focus on YTM. 
The code of GPMDP is just much nicer to work with.

**Comment by returnvisit** [comment]
agreed.
